### ISSUE TITLE
filename casing

### ISSUE THREAD

**Issue by ksjogo** [johnfn/ts2gd#49]
The Godot guide recommends snake_case for filenames: https://docs.godotengine.org/en/stable/getting_started/workflow/project_setup/project_organization.html#case-sensitivity
But it seems currently the class name is used as is to generate the file name.
Should we convert the class name to snake_case for the file name?

**Comment by johnfn** [comment]
Yes, this is a good point, though I suppose we'll have to watch out for name collisions e.g. if they have two classes, one named MyClass and one named my_class. I suppose that would be an extremely rare case though :)

**Comment by adamuso** [comment]
I chcecked and on the same page there is [Style Guide](https://docs.godotengine.org/en/stable/getting_started/workflow/project_setup/project_organization.html#style-guide) which states: 
> Use PascalCase for node names, as this matches built-in node casing.

And class names in Godot are basically names of nodes so I think current GDScript generated files names are correct.

**Comment by ksjogo** [comment]
Yes, I think we should look into moving to snake for file names.
Maybe we can check that the class names are camel when we read them and throw an Error otherwise?

**Comment by johnfn** [comment]
Check for class names to be camelCase? Do you mean snake_case? or PascalCase? Because class names should be PascalCase and then file names should be snake_case. I think it would be a little too aggressive to make class names snake_case though - especially since Godot has PascalCase class names.

**Comment by ksjogo** [comment]
Yeah, totally, check that class names are PascalCase. And then generate camel case.

**Comment by johnfn** [comment]
Ahhh gotcha, yes I'd be all in favor of us generating warnings (or errors?) if class names are not in PascalCase!
